cluster-autoscaler/processors/provreq/injector.go

```diff
@@ -116,6 +117,12 @@ func (p *ProvisioningRequestPodsInjector) GetPodsFromNextRequest(
 			p.MarkAsFailed(pr, provreqconditions.FailedToCreatePodsReason, err.Error())
 			continue
 		}
+		// Don't mark as accepted the check capacity ProvReq when batch processing is enabled.
+		// It will be marked later, in parallel, during processing the requests.
+		if pr.Spec.ProvisioningClassName == v1.ProvisioningClassCheckCapacity && p.checkCapacityBatchProcessing {
```

Comment:
I'm not a big fan of adding logic specific to particular provisioning classes here, but I guess the alternatives are just as ugly:/ 

Guess we could pass a method like "should mark PR as accepted" (similar to `isSupported`) to outsource control over this to the caller, but it probably doesn't matter much anyway.